packages/synapse-interface/pages/api/rpc/[chainId].ts

```
@@ -0,0 +1,59 @@
+
+function isDomainAllowed(headerValue: string | null): boolean {
+  if (!headerValue) return false
+  try {
+    const { hostname } = new URL(headerValue)
+    return ALLOWED_DOMAINS.some(
+      (domain) => hostname === domain || hostname.endsWith(`.${domain}`)
+    )
+  } catch {
+    return false
+  }
+}
```

ESLint

[error] Use const or class constructors instead of named functions

(prefer-arrow/prefer-arrow-functions)